### PR TITLE
gptransfer to do some extra pre-checks of users' input for partition table transfer

### PR DESCRIPTION
**Pull request by tangp3**
also updated the help doc with examples and more details.

### CODE REVIEW

File: gpMgmt/bin/gptransfer [hunk 1]
```diff
@@ -3284,27 +3288,51 @@ class GpTransfer(object):
...
                 src_table, dest_table = src_table.strip(), dest_table.strip()
-                if src_table == '' or dest_table == '':
-                    raise Exception('Can not specify empty table names')
-                if src_table == dest_table and self._options.source_host == self._options.dest_host:
+
+                if src_table == '':
+                    raise Exception('Can not specify empty source table name')
+
+                if dest_table == '':
+                    raise Exception('Can not specify empty destination table name')
```
Comment (danielgustafsson) [inline gpMgmt/bin/gptransfer]: s/can not/cannot/

File: gpMgmt/doc/gptransfer_help [hunk 2]
```diff
@@ -354,6 +354,67 @@ OPTIONS
...
+     names on each line.
+
+     Wildcard characters are not supported. Blank lines or commented lines
+     beginning with # will be ignored in the text file of -f option.
+
+     Each line lists a pair of the fully qualified name of a source leaf
+     child partition table and the fully qualified name of a destination
+     leaf child table name in this format.
+
+         src_db.scr_schema.scr_table,dest_db.dest_schema.dest_table
```
Comment (danielgustafsson) [inline gpMgmt/doc/gptransfer_help]: I assume `scr` here should be `src` ?

File: gpMgmt/doc/gptransfer_help [hunk 3]
```diff
@@ -354,6 +354,67 @@ OPTIONS
...
+             Partitioning criteria of the specified source and destination
+             leaf child partitions and child partitions above them in the
+             hierarchy (partition type and partition column).
+
+     This option is not valid with the -d, --dest-database, --drop, -F, --full,
+     --schema-only, -T or -t options.
+
+     Note: If the destination table is not empty, the table row count validation
+     fails due to row count mismatch. If the destination table is not empty, you
+     can specify the -truncate option to truncate the table before the transfer
```
Comment (danielgustafsson) [inline gpMgmt/doc/gptransfer_help]: `--truncate`

File: gpMgmt/bin/gptransfer [hunk 4]
```diff
@@ -3284,27 +3288,51 @@ class GpTransfer(object):
...
+
+                # gptransfer does not support special char '.' in db name, schema name or table name
+                if src_table.count(SCHEMA_DELIMITER) != 2:
```
Comment (danielgustafsson) [inline gpMgmt/bin/gptransfer]: Doesn't this break in case someone has been creative and used the delimiter in the tablename? While not common (I would assume) it is legal to do:
```sql
create table "with.delimiter" (a integer) distributed by (a);
```

File: gpMgmt/doc/gptransfer_help (hunk 2 above)
Comment (tangp3) [inline gpMgmt/doc/gptransfer_help]: right, to use src.

File: gpMgmt/doc/gptransfer_help (hunk 3 above)
Comment (tangp3) [inline gpMgmt/doc/gptransfer_help]: +1

File: gpMgmt/bin/gptransfer (hunk 4 above)
Comment (tangp3) [inline gpMgmt/bin/gptransfer]: Hi Daniel:
Thanks for bringing this up, GPDB does allow the dot '.' and other special chars in table name, it will be nice to provide same support in gptransfer. Since we are using plain text file as input, we need a default delimiter between schema and table name, so '.' in table name is not supported currently, this is to pre-check such scenario.

**Comment by jmcatamney** [comment]
Everything looks fine to me.

File: gpMgmt/bin/gptransfer (hunk 4 above)
Comment (danielgustafsson) [inline gpMgmt/bin/gptransfer]: You can still use a dot as the delimiter as long as you parse the textfile according to the same rules as GPDB allows for table creation. This is past the scope for this PR but it would be nice to fix that in a separate PR to close that cornercase.

Comment (tangp3) [inline gpMgmt/bin/gptransfer]: Agree, and to be improved to support the general special chars as well.
